openedx/core/lib/tests/test_cache_utils.py

```diff
@@ -297,3 +298,20 @@ def mock_wrapper(*args, **kwargs):
         result = wrapped(3)
         assert result == 2
         assert to_be_wrapped.call_count == 2
+
+
+def test_date_util_unpickling():
+    """
+    This tests is here to catch the dateutil object cache unpickling issue we faced twice on prod when attempting to
+    upgrade python-dateutil. This will help us catching such problem before it reaches production in future.
+    See https://openedx.atlassian.net/browse/BOM-2245 for more details on the issue.
+
+    The pickle is saved to the pickled_datetime.txt file using the script below.
+    >>> import dateutil.parser
+    >>> from openedx.core.lib.cache_utils import zpickle
+    >>> with open('openedx/core/lib/tests/pickled_data/pickled_datetime.txt', 'wb') as file:
+    >>>        file.write(zpickle(dateutil.parser.parse("2020-01-20T07:00:00Z")))
+    """
+    with open('openedx/core/lib/tests/pickled_data/pickled_datetime.txt', 'rb') as fp:
+        val = zunpickle(fp.read())
+        assert val < datetime.now(pytz.utc)
```

Comment:
Did you verify that this test failed with the new python-dateutil version when using a pickle created with the version on master?  Just making sure that this test case at least reproduces the problem we've already encountered.